### ISSUE TITLE
Error in names(res$query) <- rename_params(names(res$query)) 

### ISSUE THREAD

**Issue by Timlri** [artemklevtsov/RGA#23]
`get_ga()` runs for several minutes. Once it reaches 100% it halts with the error at the bottom. This is with RGA_0.4.1 from `sessionInfo()`.

``` r
> ga.data  <- get_ga(start.date  = "2013-02-01",
+                    end.date    = "2016-01-31",
+                    dimensions  = c("ga:date", "ga:pagePath"),
+                    metrics     = "ga:pageviews",
+                    max.results = NULL,
+                    fetch.by    = "day",
+                    filter      = "ga:pagePath=~/knowledgebase-articles/",
+                    #sort       = "-ga:transactions",
+                    profileId  = "ga:82227017")      # All Web Site Data (Historical)
Batch processing mode enabled.
Fetch data by day: from 2013-02-01 to 2016-01-31.
  |                                                                                                                              |   0%
Waiting for authentication in browser...
Press Esc/Ctrl + C to abort
Authentication complete.
  |==============================================================================================================================| 100%
Error in names(res$query) <- rename_params(names(res$query)) : 
  attempt to set an attribute on NULL
> 
> traceback()
3: fetch_by(path, query, by, token)
2: get_report("data/ga", query, token, fetch.by)
1: get_ga(start.date = "2013-02-01", end.date = "2016-01-31", dimensions = c("ga:date", 
       "ga:pagePath"), metrics = "ga:pageviews", max.results = NULL, 
       fetch.by = "day", filter = "ga:pagePath=~/knowledgebase-articles/", 
       profileId = "ga:82227017")

> sessionInfo()
R version 3.2.3 (2015-12-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] xlsx_0.5.7     xlsxjars_0.6.1 rJava_0.9-7    gdata_2.17.0   RGA_0.4.1     

loaded via a namespace (and not attached):
[1] httr_1.1.0      plyr_1.8.3      R6_2.1.1        tools_3.2.3     curl_0.9.4      Rcpp_0.12.2     jsonlite_0.9.19 httpuv_1.3.3   
[9] gtools_3.5.0   
```

**Comment by artemklevtsov** [comment]
Thank you for the report. Bug occurs when some or all pages is `NULL`. It should be fixed now. Please try with the latest dev-version from git.

``` r
devtools::install_github("artemklevtsov/RGA@f16b12ccc781e9c5c2ba4f40a40342e7c3c5247e")
```

**Comment by Timlri** [comment]
Hi,

I installed the update but now I get a different error at that same point. The query runs for about 20 minutes, gets to 100%, and then the error.

``` r
> ga.data  <- get_ga(start.date  = "2013-02-01",
+                    end.date    = "2016-01-31",
+                    dimensions  = c("ga:date", "ga:pagePath"),
+                    metrics     = "ga:pageviews",
+                    max.results = NULL,
+                    fetch.by    = "day",
+                    filter      = "ga:pagePath=~/knowledgebase-articles/",
+                    #sort       = "-ga:transactions",
+                    profileId  = "ga:82227017")      # All Web Site Data (Historical)
Batch processing mode enabled.
Fetch data by day: from 2013-02-01 to 2016-01-31.
  |==============================================================================================================================| 100%
Error in pages[[n]] : subscript out of bounds
> traceback()
3: fetch_by(path, query, by, token)
2: get_report("data/ga", query, token, fetch.by)
1: get_ga(start.date = "2013-02-01", end.date = "2016-01-31", dimensions = c("ga:date",
       "ga:pagePath"), metrics = "ga:pageviews", max.results = NULL,
       fetch.by = "day", filter = "ga:pagePath=~/knowledgebase-articles/",
       profileId = "ga:82227017")
```

**Comment by artemklevtsov** [comment]
The latest error fixed in 253feaa026724c22c1d8de76e892768d9178236d.

**Comment by Timlri** [comment]
That fixed it. All set now. Thanks for the quick response.
